### PR TITLE
Feat/eos core

### PR DESCRIPTION
**Pull request by gibrangul95**
cc @pancham88 @fdmirza @fariedt 
The complete implementation for Eos covering both account-lib and core

### CODE REVIEW

File: modules/account-lib/src/coin/eos/BuyRamBytesActionBuilder.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+   * @returns {string} The name of the action e.g. transfer, buyrambytes, delegatebw etc
+   */
+  actionName(): string {
+    return 'buyrambytes';
+  }
+
+  /** @inheritdoc */
+  build(builder: EosTxBuilder): EosJs.Serialize.Action {
+    const data = this.action.data;
+    if (typeof data === 'string') {
```
Comment (pancham88) [inline modules/account-lib/src/coin/eos/BuyRamBytesActionBuilder.ts:66]: Why the check for data type as string vs undefined (or not present)?

File: modules/account-lib/src/coin/eos/DeleteAuthActionBuilder.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
+import * as EosJs from 'eosjs';
```
Comment (pancham88) [inline modules/account-lib/src/coin/eos/DeleteAuthActionBuilder.ts:1]: Can you please link the documentation for these actions from EOS site? These can be included in each action documentation. It is not easy to find the documentation on EOS site?

File: modules/account-lib/src/coin/eos/BuyRamBytesActionBuilder.ts (hunk 1 above)
Comment (gibrangul95) [reply modules/account-lib/src/coin/eos/BuyRamBytesActionBuilder.ts:70]: Data is returned as a string when we deserialize the transaction without actions. Deserializing with actions returns the data as an object. Planning to move this check into the parent class in the next PR/Commit

**Comment by gibrangul95** [comment]
@pancham88 reminder for this repo

**Pull request closed without merging**